src/test/java/io/inholland/groep4/api/service/AccountsServiceTest.java

```diff
@@ -0,0 +1,159 @@
+package io.inholland.groep4.api.service;
+
+import io.inholland.groep4.api.model.Role;
+import io.inholland.groep4.api.model.User;
+import io.inholland.groep4.api.model.UserAccount;
+import io.inholland.groep4.api.security.JwtTokenProvider;
+import org.iban4j.CountryCode;
+import org.iban4j.Iban;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.BOOLEAN;
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+public class AccountsServiceTest {
+
+    @Autowired
+    private UserAccountService userAccountService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    public JwtTokenProvider jwtTokenProvider;
+
+    @Test
+    public void gettingAllAccountsShouldGiveListOfUsers() {
+        List<UserAccount> accounts = userAccountService.getAllAccounts();
+
+        assertThat(accounts).isNotEmpty();
+        assertThat(accounts.get(0).getAccountBalance()).isEqualTo(500.0);
+    }
+    @Test
+    public void gettingAccountsByIdShouldGiveAccount() {
+        Long idToFind = 1l;
+        UserAccount account = userAccountService.getAccountById(idToFind);
+
+        assertThat(account).isNotNull();
+        assertThat(account.getAccountBalance()).isEqualTo(500.0);
+    }
+
+    @Test
+    public void GettingAccountWithWrongIDShouldThrowException(){
+        Long idToFind = 100l;
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            userAccountService.getAccountById(idToFind);
+        });
+
+        assertTrue(exception.getMessage().contains("Id not found"));
+    }
+
+    @Test
+    public void gettingAccountByUserShouldGiveUserAccount(){
+        String usernameToFind = "test-user1";
+        User username = userService.findByUsername(usernameToFind);
+        List<UserAccount> userAccount = userAccountService.getAccountsByUser(username);
+
+        assertThat(userAccount).isNotEmpty();
+        assertThat(username.getUsername()).isEqualTo("test-user1");
```

Comment:
Je kijkt hier of de gebruikersnaam gelijk is aan "test-user1", maar dit hoort niet thuis in deze test.
Je moet hier een assert doen met iets van de useraccounts.